Makefile

```diff
@@ -358,6 +366,62 @@ LIBC_TOP_HALF_MUSL_SOURCES += \
         thread/wasm32/wasi_thread_start.s \
     )
 endif
+ifeq ($(THREAD_MODEL), single)
```

Comment:
There are only two thread models so this can just be `else`

Reply:
Do we _want_ to bake that in as an assumption? Or do we perhaps want to do things like having three thread models: `posix`/`stub`/`single`?

Reply:
I think we we should make this assumption up until the point where its not longer true.  For now there are only two thread models, and I think its fine for the code reflect that.